app-tracking-protection/vpn-impl/src/test/java/com/duckduckgo/mobile/android/vpn/blocklist/AppTPBlockListInterceptorApiPluginTest.kt

```diff
@@ -0,0 +1,267 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.mobile.android.vpn.blocklist
+
+import android.annotation.SuppressLint
+import com.duckduckgo.anvil.annotations.ContributesRemoteFeature
+import com.duckduckgo.app.statistics.pixels.Pixel
+import com.duckduckgo.common.test.CoroutineTestRule
+import com.duckduckgo.common.test.api.FakeChain
+import com.duckduckgo.common.test.api.InMemorySharedPreferences
+import com.duckduckgo.common.utils.DispatcherProvider
+import com.duckduckgo.data.store.api.SharedPreferencesProvider
+import com.duckduckgo.di.DaggerSet
+import com.duckduckgo.feature.toggles.api.FakeToggleStore
+import com.duckduckgo.feature.toggles.api.FeatureToggles
+import com.duckduckgo.feature.toggles.api.FeatureTogglesInventory
+import com.duckduckgo.feature.toggles.api.Toggle
+import com.duckduckgo.feature.toggles.api.Toggle.DefaultValue
+import com.duckduckgo.feature.toggles.api.Toggle.State
+import com.duckduckgo.mobile.android.vpn.feature.AppTpRemoteFeatures.Cohorts.CONTROL
+import com.duckduckgo.mobile.android.vpn.feature.AppTpRemoteFeatures.Cohorts.TREATMENT
+import com.duckduckgo.mobile.android.vpn.pixels.DeviceShieldPixelNames
+import com.duckduckgo.mobile.android.vpn.pixels.DeviceShieldPixels
+import com.duckduckgo.mobile.android.vpn.pixels.RealDeviceShieldPixels
+import com.squareup.moshi.Moshi
+import javax.inject.Inject
+import kotlinx.coroutines.withContext
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.verifyNoInteractions
+import org.mockito.kotlin.whenever
+
+// TODO: currently copied from feature-toggles-impl, but might be a better way to share this (2025-04-02)
+class RealFeatureTogglesInventory @Inject constructor(
+    private val toggles: DaggerSet<FeatureTogglesInventory>,
+    private val dispatcherProvider: DispatcherProvider,
+) : FeatureTogglesInventory {
+    override suspend fun getAll(): List<Toggle> = withContext(dispatcherProvider.io()) {
+        return@withContext toggles.flatMap { it.getAll() }.distinctBy { it.featureName() }
+    }
+
+    override suspend fun getAllTogglesForParent(name: String): List<Toggle> = withContext(dispatcherProvider.io()) {
+        return@withContext getAll().filter { it.featureName().parentName == name }
+    }
+
+    override suspend fun getAllActiveExperimentToggles(): List<Toggle> = withContext(dispatcherProvider.io()) {
+        return@withContext getAll().filter { it.getCohort() != null && it.isEnabled() }
+    }
+}
```

Comment:
Remove, why do we need this?
I think you should be good just by mocking and whenever something called return whatever list of toggles you want.